onnxruntime/core/optimizer/transpose_optimization/ort_transpose_optimization.cc

```diff
@@ -5,12 +5,35 @@
 
 #include <algorithm>
 #include "core/graph/constants.h"
+#include "core/framework/utils.h"
 #include "core/optimizer/transpose_optimization/ort_optimizer_utils.h"
 
 using namespace onnx_transpose_optimization;
 
 namespace onnxruntime {
 
+static bool EPAwareHandleResize(HandlerArgs& args) {
+  // Whilst Resize is not technically layout sensitive, execution providers typically implement handling for only one
+  // layout. Due to that, only push a Transpose through a Resize once it is assigned and we know it's being handled
+  // by an EP that supports multiple layouts. Currently that's the CPU and XNNPACK EPs.
+  const auto ep_type = args.node.GetExecutionProviderType();
+  if (ep_type == kCpuExecutionProvider || ep_type == kXnnpackExecutionProvider) {
+    // allow NCHW <-> NHWC for now. not clear any other sort of transpose has a valid usage in a real model
+    int64_t rank_int = gsl::narrow_cast<int64_t>(args.perm.size());
```

Comment:
nit: doesn't seem to require a narrow_cast from size_t to int64_t.

Reply:
IIUC int64_t is signed and size_t isn't, so for 64-bit they have different ranges even though they're the same number of bits and the 'narrowing' here is dealing with values outside that range (which will never happen as the size of the vector will never be negative so the compiler doesn't know that).

Reply:
Right, but this `rank_int` is only just compared to the literal `4`, so making it signed for this comparison seems unnecessary (but harmless).